draft-ietf-tls-ctls.md

```diff
@@ -450,15 +548,69 @@ framing, except for two changes:
               case finished:              Finished;
               case new_session_ticket:    NewSessionTicket;
               case key_update:            KeyUpdate;
+              case request_connection_id: RequestConnectionId;
+              case new_connection_id:     NewConnectionId;
           };
-      } Handshake;
+      } CTLSHandshake;
 ~~~~
 
+Each `CTLSHandshake` MUST be conveyed as a single `CTLSPlaintext.fragment`
+or `CTLSCiphertext.encrypted_record`, and is therefore limited to a maximum
+length of `2^16-1`.  When operating over UDP, large `CTLSHandshake` messages
+will also require the use of IP fragmentation, which is sometimes
+undesirable.  Operators can avoid these concerns by setting
+`template.handshake_framing = 1`.
+
+#### Retransmission
+
+Like DTLS, Datagram cTLS requires a retransmission mechanism when operating over
+a lossy transport.  When `handshake_framing` is true, Datagram cTLS uses the same
+ACK and retransmission system as the corresponding version of DTLS.  However, when
+`template.handshake_framing` is false, retransmissions work slightly differently:
+:
+* `ACK.sequence_number` is computed as the number of messages in the handshake
+transcript since the last KeyUpdate, starting with the ClientHello at `sequence_number = 1`.
+* Retransmissions do not increment the `sequence_number`.
+* Each message type can only appear once from each sender in the handshake.  Recipients MUST ignore any duplicated messages.
+* Messages within a flight are placed in canonical order by the recipient.
+
+These rules are sufficient to ensure that the handshake terminates, and both parties
+agree on the sequence of messages that are received, even if some records are
+dropped or duplicated by the network.
```

Comment:
This is going to require some thinking on my part to be sure this works.

Reply:
I've dropped this entirely due to changes in the `handshakeFraming` PR.